src/data/list/defs.lean

```diff
@@ -167,9 +167,54 @@ def find (p : α → Prop) [decidable_pred p] : list α → option α
 | (a::l) := if p a then some a else find l
 
 /-- `mfind tac l` returns the first element of `l` on which `tac` succeeds, and fails otherwise. -/
-def mfind {α} {m : Type → Type u} [monad m] [alternative m] (tac : α → m unit) : list α → m α :=
+def mfind {α} {m : Type u → Type v} [monad m] [alternative m] (tac : α → m punit) : list α → m α :=
 list.mfirst $ λ a, tac a $> a
 
+/-- `mbfind' p l` returns the first element `a` of `l` for which `p a` returns
+true. `mbfind'` short-circuits, so `p` is not necessarily run on every `a` in `l`. -/
+def mbfind' {m : Type u → Type v} [monad m] {α : Type u} (p : α → m (ulift bool))
+  : list α → m (option α)
+| [] := pure none
+| (x :: xs) := do
+  ⟨px⟩ ← p x,
+  if px then pure (some x) else mbfind' xs
+
+section
+
+variables {m : Type → Type v} [monad m]
+
+/-- A variant of `mbfind'` with more restrictive universe levels. -/
+def mbfind {α} (p : α → m bool) (xs : list α) : m (option α) :=
+xs.mbfind' (functor.map ulift.up ∘ p)
+
+/-- `mbany p as` returns true iff `p` returns true for any element of `l`.
+`mbany` short-circuits, so if `p` returns true for any element of `l`, later
+elements are not checked. -/
+-- Implementing this via `mbfind` would give us less universe polymorphism.
+def many {α : Type u} (p : α → m bool) : list α → m bool
+| [] := pure false
+| (x :: xs) := do px ← p x, if px then pure tt else many xs
+
+/-- `mball p as` returns true iff `p` returns true for all elements of `l`.
+`mball` short-circuits, so if `p` returns false for any element of `l`, later
+elements are not checked. -/
+def mall {α : Type u} (p : α → m bool) (as : list α) : m bool :=
```

Comment:
The name and docstring don't match here.

Reply:
Fixed.